ods-devenv/scripts/deploy.sh

```diff
@@ -0,0 +1,472 @@
+    docker container cp mysql-connector-java-8.0.20.jar bitbucket:/var/atlassian/application-data/bitbucket/lib/mysql-connector-java-8.0.20.jar
+    rm mysql-connector-java-8.0.20.jar
+}
+
+#######################################
+# Initialize bitbucket database. Requires a database service to be running.
+# Globals:
+#   atlassian_bitbucket_db_name
+#   atlassian_mysql_container_name
+#   atlassian_mysql_port
+#   atlassian_mysql_version
+# Arguments:
+#   n/a
+# Returns:
+#   None
+#######################################
+function initialize_atlassian_bitbucketdb() {
+    local mysql_ip=$(docker inspect -f '{{.NetworkSettings.IPAddress}}' ${atlassian_mysql_container_name})
+    echo "Setting up bitbucket database on ${mysql_ip}:${atlassian_mysql_port}."
+    echo "jiradbrpwd" | docker container run -i --rm mysql:${atlassian_mysql_version} mysql -h ${mysql_ip} -u root -p -e \
+        "create database ${atlassian_bitbucket_db_name} character set utf8 collate utf8_bin; \
+        GRANT SELECT,INSERT,UPDATE,DELETE,CREATE,DROP,REFERENCES,ALTER,INDEX,CREATE TEMPORARY TABLES on bitbucketdb.* TO 'bitbucket_user'@'%' IDENTIFIED BY 'bitbucket_password'; \
+        flush privileges;"
+}
+
+#######################################
+# Overwrites existing bitbucketdb data files with a backup from a fresh bitbucketdb incl
+# timebomb licenses.
+# Globals:
+#   n/a
+# Arguments:
+#   n/a
+# Returns:
+#   None
+#######################################
+function restore_atlassian_bitbucketdb_with_license() {
+    local target_dir=/home/${USER}/mysql_data
+    sudo rm -rf ${target_dir}/bitbucketdb
+    sudo tar xzf ${BASH_SOURCE%/*}/../atlassian/bitbucketdb.tar.gz -C ${target_dir}/..
+}
+
+#######################################
+# The automated ODS setup requires some repositories to be existing in the
+# local bitbucket installation. This function takes care of it.
+# Globals:
+#   n/a
+# Arguments:
+#   n/a
+# Returns:
+#   None
+#######################################
+function create_empty_ods_repositories() {
+    # The repository list in the next line can be modified to project specific needs.
+    # For each of the listed names, a repository will be created in the local bitbucket
+    # instance under the OPENDEVSTACK project. The list should be synced with the repo
+    # list in ods-core/ods-setup/repos.sh.
+    for repository in ods-core ods-quickstarters ods-jenkins-shared-library ods-provisioning-app; do
+        echo "Creating repository ${repository} on http://localhost:${atlassian_bitbucket_port}."
+        curl -X POST --user openshift:openshift http://localhost:${atlassian_bitbucket_port}/rest/api/1.0/projects/opendevstack/repos \
+            -H "Content-Type: application/json" \
+            -d "{\"name\":\"${repository}\", \"scmId\": \"git\", \"forkable\": true}" | jq .
+    done
+}
```

Comment:
I would love to see this function extracted into its own shell script, and hooked up from the `Makefile` as a new target so that users following the installation docs do not need to manually create those.

On a side note, `ods-provisioning-app` is not required anymore for a default install - as we pull the image from Docker Hub now.

Reply:
@michaelsauter  - for the dev env - we should provision the quickstarters for docgen and prov app as well :) which is a great test after all :)

Reply:
this is addressed via the call to bitbucket.sh and repos.sh :)